src/resmom/linux/alps.c

```diff
@@ -3537,7 +3538,7 @@ parse_nidlist_char_data(ud_t *d, const char *s, int len)
 	 * Check if the current rangelist of Nodes is of type KNL and that such Nodes
 	 * are in "batch" mode and in the "up" state.
 	 */
-	if (!exclude_from_KNL_processing(d->current_sys.node_group)) {
+	if (!exclude_from_KNL_processing(d->current_sys.node_group, 0)) {
```

Comment:
Since you aren't checking the state here, please fix the above comment to match the behavior.  And add an explanation as to why you don't want to check the KNL node state here.